src/engine/entity/Entity.hpp

```diff
@@ -96,5 +149,6 @@ class Entity {
 
   private:
     entity_id_type _entity;
+    inline static std::unordered_map<std::type_index, std::function<void(Registry &)>> temporaryComponent = {};

```

Comment:
name the map tagComponents as well

Reply:
It's not a tag, I renamed `RemoveTags` to `RemoveTemporaryComponents`.